sql/create.go

```diff
@@ -142,6 +142,8 @@ func (p *planner) CreateIndex(n *parser.CreateIndex) (planNode, error) {
 		return nil, err
 	}
 
+	// Mark transaction as operating on the system DB.
+	p.txn.SetSystemDBTrigger()
```

Comment:
ok, but how do I make sure this gets passed through the txn_coord_sender? If I don't persist this as part of the transaction object and make sure it gets properly passed, this will fail.
eg: see `TestSystemDBGossip` which uses a regular kv client. This required the modifications in `proto/data.go` to propagate the field when constructing a new transaction.